dev/tools/localization/gen_l10n.dart

```diff
@@ -515,22 +525,40 @@ class LocalizationsGenerator {
     return !(statString[1] == 'w' || statString[4] == 'w' || statString[7] == 'w');
   }
 
-  /// Sets the reference [Directory] for [l10nDirectory].
+  /// Sets the reference [Directory] for [inputDirectory].
   @visibleForTesting
-  void setL10nDirectory(String arbPathString) {
-    if (arbPathString == null)
-      throw L10nException('arbPathString argument cannot be null');
-    l10nDirectory = _fs.directory(arbPathString);
-    if (!l10nDirectory.existsSync())
+  void setInputDirectory(String inputPathString) {
+    if (inputPathString == null)
+      throw L10nException('inputPathString argument cannot be null');
+    inputDirectory = _fs.directory(inputPathString);
+    if (!inputDirectory.existsSync())
       throw FileSystemException(
-        "The 'arb-dir' directory, $l10nDirectory, does not exist.\n"
+        "The 'input-dir' directory, $inputDirectory, does not exist.\n"
         'Make sure that the correct path was provided.'
       );
 
-    final FileStat fileStat = l10nDirectory.statSync();
+    final FileStat fileStat = inputDirectory.statSync();
+    if (_isNotReadable(fileStat) || _isNotWritable(fileStat))
+      throw FileSystemException(
+        "The 'input-dir' directory, $inputDirectory, doesn't allow reading and writing.\n"
+        'Please ensure that the user has read and write permissions.'
+      );
+  }
+
+  /// Sets the reference [Directory] for [outputDirectory].
+  @visibleForTesting
+  void setOutputDirectory(String outputPathString) {
+    if (outputPathString == null)
+      throw L10nException('outputPathString argument cannot be null');
+    outputDirectory = _fs.directory(outputPathString);
+    if (!outputDirectory.existsSync()) {
```

Comment:
Ideally, we wouldn't create the output directory until all the input validity checks had succeeded, i.e. until we were about to start writing files.

Reply:
I tried addressing this with my latest commits. PTAL